src/main/java/io/tiledb/spark/TileDBBatch.java

```diff
@@ -297,14 +381,16 @@ private Pair<List<List<Range>>, Class> buildRangeFromFilter(
         second = nonEmptyDomain.get(f.attribute()).getSecond();
       else second = getMaxValue(f.value().getClass());
       int columnIndex = this.tileDBReadSchema.getColumnId(f.attribute()).get();
-      ranges
-          .get(columnIndex)
-          .add(
-              new Range(
-                  new Pair<>(
-                      addEpsilon(
-                          (Number) f.value(), this.tileDBReadSchema.columnTypes.get(columnIndex)),
-                      second)));
+      Number lowerBound =
+          addEpsilon((Number) f.value(), this.tileDBReadSchema.columnTypes.get(columnIndex));
```

Comment:
@DimitrisStaratzis I guess it's pre-existing, but could you please remind me why we do this? I guess it is to expand the query range, but tiledb ranges are already endpoint-inclusive... And add a comment inline at the function body definition explaining why.

Reply:
It's not my code, but it is because the ranges that were used before applying the QC were inclusive and depending on the datatype of the ranges they add an epsilon value which is the smallest value for the given type. This part of the code is used only for the dimensions currently, and will be deleted when the QC supports dimensions. I think it does not yet.

Reply:
Ok, see - this is pushdown of a `greaterThan` operator to a range query.